File: matlab_code/oldcode/homework3_2.m
%2x+1=e^x«–œﬂ∑®
f=@(x)2*x+1-exp(x);
df=@(x)2-exp(x);
x=linspace(-0.8,2.6,1001);
plot(x,f(x));
hold on;plot(x,zeros(1,1001));

x=0.8;
for i=1:30
    t=x-f(x)/df(x);
    
    hold on;plot([x,t],[f(x),0]);pause(1)
    
    x=t;
    
    hold on;plot([x,x],[f(x),0],'--');pause(1);
end

File: anacondacode/tensorflow_environment/old_code/计算方法/homework3_2.py
# -*- coding: utf-8 -*-
"""
Created on Sat Sep 21 12:04:48 2019

@author: LJZ
"""

#2x+1=e^x切线法
import numpy as np
import matplotlib.pyplot as plt
import math
e=math.e
f=lambda x: 2*x+1-e**x
df=lambda x:2-e**x
x=np.linspace(-0.8,2.6,1001)
plt.plot(x,f(x))
#hold on
plt.plot(x,np.zeros(x.shape))

x=0.8
for i in range(1,31):
    t=x-f(x)/df(x)
    
    #hold on
    plt.plot([x,t],[f(x),0])
    #pause(1)
    
    x=t
    
    #hold on
    plt.plot([x,x],[f(x),0],'--')
    #pause(1)
